ucfg_test.go

```diff
@@ -61,3 +68,242 @@ func TestFlattenKeys(t *testing.T) {
 		})
 	}
 }
+
+func TestDetectCyclicReference(t *testing.T) {
+	tests := []struct {
+		title    string
+		cfg      map[string]interface{}
+		expected string
+	}{
+		{
+			title: "direct cyclic reference",
+			cfg: map[string]interface{}{
+				"top.reference": "${top.reference}",
+			},
+			expected: "cyclic reference detected for key: 'top.reference' accessing 'top.reference'",
+		},
+		{
+			title: "compound reference",
+			cfg: map[string]interface{}{
+				"top.reference": "hello ${top.reference}",
+			},
+			expected: "cyclic reference detected for key: 'top.reference' accessing 'top.reference'",
+		},
```

Comment:
```
{
  title: "reference to parent",
  cfg: map[string]interface{}{
    "top.reference": "${top}"
  }
  expected: ...
}
```

In tests I don't try to match the actual error message, but the presence of an error (and potentially the error type). That's what `Error` is for, for example.

For tests on actual error messages, see `error_test.go`. The `raiseX` functions in `errror.go` are used to generate the actual errors.

But then some internal code in go-ucfg still uses `error` :(